### ISSUE TITLE
Nightly TicWatch Pro / MM2 / Pixel 2 XL Android 10 issues after install

### ISSUE THREAD

**Issue by Vigilo-Rob** [NightscoutFoundation/xDrip#1195]
Hi All,

Finding an issue post 31st Dec 19 release.

With 31st Dec - Android 10 works ok as long as WearOS is set as:
  Enable Wear Collection Service,
  Force Wear Collection Service
If Force Wear is not set, after 6-8 hours I get connection issues and have to turn Android integration off, and back on again once my phone has connected.

With 9th Jan connection to MM2 dropped after a few minutes, battery drain on watch was approx 1/3 higher - so reverted to 31st Dec

Just tried 21st Jan - TicWatch pro is pretty much unusable (incredibly slow response to screen presses, would connect to MM2 but dropped connections after 1st reading), screen goes completely blank and requires button presses to show data (again very very slowly).

Tried reboots of Ticwatch - no change.

Has anyone else had similar issues?

This is a life changing development, and wouldn't want to be without xDrip (my wife sleeps much better now!) - thank you for all your hard work.

Rob.

**Comment by DavidAG2000** [comment]
I'm using a MM2 and a Google Pixel 2 XL with Android 10, paired with a Fossil Gen 5, and for me it's all working fine, but with the latest version of xDrip+ (they managed to make watches work as collector with MM2 in these latest versions). Have a try.

https://github.com/NightscoutFoundation/xDrip/releases/tag/2020.01.21

**Comment by Vigilo-Rob** [comment]
@DavidAG2000  - many thanks for the response

Did some more digging as was having issues with disconnects even when rolled back to 31st Dec release - realised not uninstalled on my phone before installing the new nightly (think I read that its advised to uninstall on phone and watch for each release).

Once it was installed cleanly - the 21st Jan nightly is working well - only had one disconnect from Ticwatch over night (logs showing "Bluetooth Audio Disconnected" so guessing BT drops the connection for some reason.

Looking at the error log I'm getting a lot of "Error getting data from getLatestBgForXMinutes Returning" followed by 12 x "Error for time 5696 Existing Value 1225!= 1222" with different value errors.

Any ideas whats causing either of those issues - or how to work round them, the BT connection drops is more problematic as I have to re-start the collector on the watch (about 5am this morning).

Thanks again. :-)

**Comment by DavidAG2000** [comment]
@Vigilo-Rob 
I'm glad I helped you with your issue 🙂
How did you manage to finally solve your problem?

**Comment by Vigilo-Rob** [comment]
@DavidAG2000 

A great deal of frowning..... :-)

I reverted to the 31st Dec release which worked for an hour or so, then started giving dropped connections and unpredictable results when connecting my watch.

I remember reading somewhere (either threads or documentation) that we should uninstall and re-install rather than doping an in-place upgrade, so I gave it a go.

Thanks for your prod in the right direction - very useful to know someone out there is working ok on a very similar set-up.

Cheers,

Rob.
